src/client/app/components/BarControlsComponent.tsx

```diff
@@ -17,26 +16,83 @@ import TooltipMarkerComponent from './TooltipMarkerComponent';
  */
 export default function BarControlsComponent() {
 	const dispatch = useAppDispatch();
+
+	// The min/max days allowed for user selection
+	const MIN_BAR_DAYS = 1;
+	const MAX_BAR_DAYS = 366;
+	// Special value if custom input for standard menu.
+	const CUSTOM_INPUT = '-99';
+
+	// This is the current bar interval for graphic.
 	const barDuration = useAppSelector(selectBarWidthDays);
 	const barStacking = useAppSelector(selectBarStacking);
-	const [showSlider, setShowSlider] = React.useState<boolean>(false);
-	const [sliderVal, setSliderVal] = React.useState<number>(barDuration.asDays());
-
-	const toggleSlider = () => {
-		setShowSlider(showSlider => !showSlider);
-	};
+	// should render as custom on initial render if not found in controlled values.
+	// This can happen with a chart link of custom bar duration.
+	const shouldRenderCustom = !(['1', '7', '28'].find(days => days == barDuration.asDays().toString()));
+	// Holds the value of standard bar duration choices used so decoupled from custom and
+	// also to allow special value for custom choice.
+	const [barDays, setBarDays] = React.useState<string>(shouldRenderCustom ? CUSTOM_INPUT : barDuration.asDays().toString());
+	// Holds the value during custom bar duration input so only update graphic when done entering and
+	// separate from standard choices.
+	const [barDaysCustom, setBarDaysCustom] = React.useState<number>(barDuration.asDays());
+	// True if custom bar duration input is active.
+	const [showCustomBarDuration, setShowCustomBarDuration] = React.useState<boolean>(shouldRenderCustom);
 
 	const handleChangeBarStacking = () => {
 		dispatch(graphSlice.actions.changeBarStacking());
 	};
 
-	const handleSliderChange = (value: number) => {
-		setSliderVal(value);
+	// Update when the custom input bar duration is shown/hidden.
+	React.useEffect(() => {
+		if (showCustomBarDuration) {
+			// It changed to true so set the input value as the current barDuration
+			// so shows in custom field at start of input.
+			setBarDaysCustom(barDuration.asDays());
+		}
+		// If false then hiding custom so value does not matter.
+	}, [showCustomBarDuration]);
+
```

Comment:
Further testing revealed that there was issues keeping redux, and react level state in sync. Similar to the chart link issue, using the history buttons would leave the custom input and dropdown out of sync. My earlier suggestion was useful for syncing state to the initial render, but using the history feature would again leave the component out of sync. 
So moving that logic from initial render into a use effect, will keep in sync when this component's state is altered externally. E.g. not when interacting with the component.
I've tested this, and it works as expected. If you concur this Pr looks good to go. 
```suggestion

	// Holds the value of standard bar duration choices used so decoupled from custom and
	// also to allow special value for custom choice.
	const [barDays, setBarDays] = React.useState<string>(barDuration.asDays().toString());
	// Holds the value during custom bar duration input so only update graphic when done entering and
	// separate from standard choices.
	const [barDaysCustom, setBarDaysCustom] = React.useState<number>(barDuration.asDays());
	// True if custom bar duration input is active.
	const [showCustomBarDuration, setShowCustomBarDuration] = React.useState<boolean>(false);

	const handleChangeBarStacking = () => {
		dispatch(graphSlice.actions.changeBarStacking());
	};

	// Keeps react-level state, and redux state in sync.
	// Two different layers in state may differ especially when externally updated (chart link, history buttons.)
	React.useEffect(() => {
		// Assume Value is valid  since it is comming from state.
		// Do not allow bad values in state.
		const isCustom = !(['1', '7', '28'].find(days => days == barDuration.asDays().toString()));
		setShowCustomBarDuration(isCustom);
		setBarDaysCustom(barDuration.asDays());
		setBarDays(isCustom ? CUSTOM_INPUT : barDuration.asDays().toString());
	}, [barDuration]);
```